Hw3.cpp

IT-kakaya opened the pull request.

IT-kakaya committed:
Create taskChess.cpp

IT-kakaya committed:
Create task2.cpp

Почему-то получается так, что вводится только одно число, и с enter сразу выводится результат

dmitrii-kruglov commented on task2.cpp:
```diff
@@ -0,0 +1,28 @@
+        do{
+            if (c=='-'){
+                sign=false;
+                
```
BP: лишняя строка

dmitrii-kruglov commented on task2.cpp:
```diff
@@ -0,0 +1,28 @@
+                sign=true;
+            }
+        }while(c=='\n');
+        a=c-'0'+a*10;
```
слишком много в main, нужно разбить на функции

dmitrii-kruglov commented on task2.cpp:
```diff
@@ -0,0 +1,28 @@
+            else{
+                sign=true;
+            }
+        }while(c=='\n');
```
внутри цикла c не изменяется. Бесконечный?

dmitrii-kruglov commented on taskChess.cpp:
```diff
@@ -0,0 +1,46 @@
+    cin>>a;
+    cin>>b;
+    cout<<"Cells you can go to:"<<endl;
+    for(i=1; i<=8; i++){
```
слишком большая функция, необходимо разбить на несколько

dmitrii-kruglov reviewed (changes requested):
Пожалуйста, делайте для одного задания один пул реквест

IT-kakaya committed:
Update taskChess.cpp

Переделано (из мейна сделаны функции)

IT-kakaya committed:
Update task2.cpp

Переделано с учетом изменения условия задачи (остановка ввода после ввода "0")

dmitrii-kruglov commented on taskChess.cpp:
```diff
@@ -0,0 +1,44 @@
+#include <iostream>
+using namespace std;
+
+char Step1(char a, int i){
```
неужели у вас компилируется без ошибок? Покажите на семинаре, пожалуйста.

dmitrii-kruglov commented on task2.cpp:
```diff
@@ -0,0 +1,27 @@
+int PositiveNumbers(int a){
+    int negative = 0;
+    if (a < 0){
+        negative = negative + 1;
```
Слишком сложно... Почему нельзя написать просто - negative = 1?

dmitrii-kruglov commented on task2.cpp:
```diff
@@ -0,0 +1,27 @@
+#include <iostream>
+using namespace std;
+
+int PositiveNumbers(int a){
```
название же не совпадает со смыслом? Возвращает же 1 если отрицательное, и 0 если положительное..

dmitrii-kruglov commented on task2.cpp:
```diff
@@ -0,0 +1,27 @@
+    cout << "Enter numbers. If you want to stop, enter 0:" << endl;
+    while (a != 0){
+        cin>>a;
+        PositiveNumbers(a);
```
зачем два раза запускать функцию?

dmitrii-kruglov reviewed (changes requested):
Нужно дорабатывать

IT-kakaya committed:
Update task2.cpp

Все исправила, название функции тоже поменяла

IT-kakaya committed:
Update taskChess.cpp

Раньше значения тоже выводились, но после предупреждения о том, что функция ничего не возвращает. Сейчас я добавила return, код работает.

dmitrii-kruglov commented on taskChess.cpp:
```diff
@@ -0,0 +1,46 @@
+        cout << char(a-1) << i << endl; 
+    }
+    if (char(a) < 'g'){
+        cout << char(a+1) << i << endl;
```
Пробелы...

dmitrii-kruglov reviewed (approved):
Принято

IT-kakaya committed:
Update taskChess.cpp

Доделала пробелы
